lualib/resty/events/callback.lua

```diff
@@ -20,34 +20,34 @@ local _M = {
 local _MT = { __index = _M, }
 
 function _M.new()
-    local self = {
+    return setmetatable({
         _callbacks = {},
         _funcs = {},
         _counter = 0,
-    }
-
-    return setmetatable(self, _MT)
+    }, _MT)
 end
 
 local function get_callback_list(self, source, event)
-    local _callbacks = self._callbacks
-
-    if not _callbacks[source] then
-        _callbacks[source] = {}
-    end
```

Comment:
Seems we only create table if not exist. Do we need to split?

The refactor's main work is to move the creation job to a separate function.

Reply:
Yes, and the whole idea here is not to create when not exits. E.g. worker x has not subscribed event on target `doge` with event `blaa`, now it still creates tables for it. The change avoid worker x that has not subscribed `doge` with `blaa` to not create tables for it.

Reply:
Make sense. We only check if we need to create table in the subscribe function, not in the do event function.